### PR TITLE
Refactor luigi config

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:

- Separate navigation model computation from  Luigi initialisation 
- Use Luigi API for token handling
- Fetch namespaces (to context switcher) using gql (and delete `refreshContextSwitcher` handling)
- Code refactoring and cleanup
- Fixed current namespace context reloading

**Related issue(s)**
Fixes https://github.com/kyma-project/console/issues/1269
Fixes https://github.com/kyma-project/console/issues/1595
Fixes https://github.com/kyma-project/console/issues/954

### CODE REVIEW

File: core/src/luigi-config/navigation/navigation-data-init.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+      {
+        icon: 'settings',
+        label: 'Settings',
+        link: '/home/settings'
+      }
+    ]
+  }
+};
+
+export function getNavigationData() {
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply core/src/luigi-config/navigation/navigation-data-init.js]: By design, this one is the main function of the navigation-data-init.
thats the purpose of the file - contain all the logic for fetching data behind the navigation model.

It still long, agree. But much shorter than before :)

File: core/src/luigi-config/navigation/navigation-data-init.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  var staticNodes = getStaticChildrenNodesForNamespace(namespace)
+
+  return Promise.all([
+    getMicrofrontends(namespace),
+    Promise.resolve(clusterMicrofrontendNodesForNamespace)
+  ])
+    .then(function(values) {
+      var nodeTree = [...staticNodes];
+      values.forEach(function(val) {
+        nodeTree = [].concat.apply(nodeTree, val);
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply core/src/luigi-config/navigation/navigation-data-init.js]: doesnt wotrk

File: core/src/luigi-config/navigation/navigation-helpers.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import { config } from '../config';
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply core/src/luigi-config/navigation/navigation-helpers.js]: fixed

File: core/src/luigi-config/navigation/navigation-data-init.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    });
+  });
+}
+
+function getNamespaces() {
+  const options = {
+    showSystemNamespaces : localStorage.getItem('console.showSystemNamespaces') === 'true',
+    withInactiveStatus : false,
+  }
+  return fetchFromGraphQL(GET_NAMESPACES, options, true)
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply core/src/luigi-config/navigation/navigation-data-init.js]: fixed

File: core/src/luigi-config/luigi-config.js [hunk 5]
```diff
@@ -389,335 +30,86 @@ window.addEventListener('message', e => {
...
-  .catch(err => {
-    console.error('Config Init Error', err);
-  });
+    }
+  }
+};
+Luigi.setConfig(luigiConfig);
 
 window.addEventListener('message', e => {
   if (e.data.msg && e.data.msg === 'console.quotaexceeded') {
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply core/src/luigi-config/luigi-config.js]: done

File: core/src/luigi-config/luigi-event-handlers.js [hunk 6]
```diff
@@ -2,9 +2,8 @@ import {
   setLimitExceededErrorsMessages
 } from './navigation/navigation-helpers';
 
-window.addEventListener('message', e => {
-  if (e.data.msg && e.data.msg === 'console.quotaexceeded') {
-    const namespace = e.data.namespace;
+export function onQuoteExceed(eventData) {
```
Comment (parostatkiem) [inline core/src/luigi-config/luigi-event-handlers.js]: `onQuotaExceed` ?

File: core/src/luigi-config/navigation/navigation-helpers.js [hunk 7]
```diff
@@ -139,3 +139,12 @@ function getCorrespondingNamespaceLocation(namespaceName) {
   }
   return namespaceName + '/' + addressTokens.slice(4).join('/');
 }
+
+export function parseJWT(token){
+  var base64Url = token.split('.')[1];
+  var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  var jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
```
Comment (parostatkiem) [inline core/src/luigi-config/navigation/navigation-helpers.js]:
```suggestion
  var jsonPayload = decodeURIComponent(atob(base64).split('').map(c =>
 '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
```

**Pull request merged**